### PR TITLE
ENH: LightInterface for slits

### PR DESCRIPTION
**Pull request by teddyrendahl**
Added tests for slits methods as well as making lightpath compliant. Used an interesting schema to handle the strange-ness of the slits interaction with the beam. See the notes in the class doc-string.

### CODE REVIEW

File: pcdsdevices/epics/slits.py [hunk 1]
```diff
@@ -44,13 +66,26 @@ class Slits(IocDevice):
     close_cmd = Component(EpicsSignal, ":CLOSE")
     block_cmd = Component(EpicsSignal, ":BLOCK")
 
+    #Nominal
+    nominal_aperature = 5.0
```
Comment (ZLLentz) [inline pcdsdevices/epics/slits.py]: Any thoughts on making this a parameter? Some of the scientists open the slits to 10mm. I'm not sure it really matters.

File: pcdsdevices/epics/slits.py [hunk 2]
```diff
@@ -106,10 +207,28 @@ def unstage(self):
         return super().unstage()
 
     def open(self):
+        """
+        Open the slits to have a 50mm aperature on each side
```
Comment (ZLLentz) [inline pcdsdevices/epics/slits.py]: docstring typo?

File: pcdsdevices/epics/slits.py [hunk 3]
```diff
@@ -44,13 +66,26 @@ class Slits(IocDevice):
...
+            read_attrs = ['xcenter', 'xwidth', 'ycenter', 'ywidth']
         super().__init__(prefix, ioc=ioc, read_attrs=read_attrs, name=name,
                          **kwargs)
 
+        #Subscribe to changes in aperature
+        self.xwidth.readback.subscribe(self._aperature_changed,
+                                       run=False)
+        self.ywidth.readback.subscribe(self._aperature_changed,
+                                       run=False)
+
```
Comment (ZLLentz) [inline pcdsdevices/epics/slits.py]: We need to go back at some point and clean up all of our classes that subscribe during the init, when they don't really need to subscribe unless someone else subscribes to them. I have a feeling it's bad for pyepics to be unnecessarily monitoring things.

Comment (teddyrendahl) [reply pcdsdevices/epics/slits.py]: Maybe correct. Not sure I want to worry about it until we actually see an issue.

File: pcdsdevices/epics/slits.py (hunk 2 above)
Comment (teddyrendahl) [reply pcdsdevices/epics/slits.py]: MFX is all set to 50 (which is a terrible value). It is just the `OPEN.DOL*` fields that control the value so this is probably set to a more sane value in hutches where people use this record. I'll edit the docstring to not specify a value

File: pcdsdevices/epics/slits.py (hunk 1 above)
Comment (teddyrendahl) [reply pcdsdevices/epics/slits.py]: 👍

**Pull request merged**